### ISSUE TITLE
`padding-line-between-statements` function overload support

### ISSUE THREAD

**Issue by seahindeniz** [eslint-stylistic/eslint-stylistic#190]
### Clear and concise description of the problem

I would like to avoid adding an empty line before function overloads and the function itself. Currently, there's no way to specify overloads as a statement in the rules options

Actual
```ts
function foo(): void;
function foo(param0?: boolean): boolean;

function foo(param0?: boolean) {
  //
}
```

### Suggested solution

Since `interface` and `type` has their own statement key, having `function-overload` or `overload` `fn-overload`(👎🏻 ) as an alias would be better.

Expected:
```ts
function foo(): void;
function foo(param0?: boolean): boolean;
function foo(param0?: boolean) {
  //
}
```

### Additional context

```js
{
    '@stylistic/padding-line-between-statements': [
        'error',
        {
            blankLine: 'always',
            next: ['multiline-block-like'],
            prev: ['*'],
        },
    ]
}
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
